File: api/career_map.py
# Career roadmap - create a roadmap of your dream job role
import os
import google.generativeai as genai
from dotenv import load_dotenv

load_dotenv()

genai.configure(api_key=os.environ["GEMINI_API_KEY"])


def get_roadmap(role):
    # Create the model
    generation_config = {
        "temperature": 1,
        "top_p": 0.95,
        "top_k": 40,
        "max_output_tokens": 8192,
        "response_mime_type": "text/plain",
    }

    model = genai.GenerativeModel(
        model_name="gemini-1.5-flash",
        generation_config=generation_config,
        system_instruction=
        "I will give you my dream Job Role and you must create a proper roadmap that i must follow to achieve it.",
    )

    chat_session = model.start_chat(history=[])
    prompt = f"{role}"
    response = chat_session.send_message(prompt)

    print(response.text)

File: src/feature_functions/CareerMap.js
import { GoogleGenerativeAI } from "@google/generative-ai";

// dotenv.config();

const genAI = new GoogleGenerativeAI("AIzaSyCDz4gjhidaJvdzPjYHxdo1o0CAqiqll9k");

export async function getRoadmap(role) {
    const model = genAI.getGenerativeModel({
        model: "gemini-1.5-flash",
        systemInstruction: "I will give you my dream Job Role and you must create a proper roadmap that i must follow to achieve it. the output should be in HTML format",
    });

    const generationConfig = {
        temperature: 1,
        topP: 0.95,
        topK: 40,
        maxOutputTokens: 8192,
    };

    try {
        const chatSession = model.startChat({
            history: [],
            generationConfig,
        });

        const result = await chatSession.sendMessage(role);
        console.log(result.response.text());
        return result.response.text();
    } catch (error) {
        console.error("Error:", error);
        return "An error occurred while generating the roadmap.";
    }
}

// Example usage
// getRoadmap("Machine Learning Engineer");